src/spatialdata/_core/operations/map.py

```diff
@@ -0,0 +1,122 @@
+from xarray import DataArray
+
+from spatialdata.models._utils import get_axes_names, get_channels, get_raster_model_from_data_dims
+from spatialdata.models.models import Labels2DModel, Labels3DModel, get_model
+from spatialdata.transformations import get_transformation
+
+__all__ = ["map_raster"]
+
+
+def map_raster(
+    data: DataArray | DataTree,
+    func: Callable[[da.Array], da.Array],
+    fn_kwargs: Mapping[str, Any] = MappingProxyType({}),
+    blockwise: bool = True,
+    depth: int | tuple[int, ...] | dict[int, int] | None = None,
+    chunks: tuple[tuple[int, ...], ...] | None = None,
+    c_coords: Iterable[int] | Iterable[str] | None = None,
+    dims: tuple[str, ...] | None = None,
+    transformations: dict[str, Any] | None = None,
+    **kwargs: Any,
+) -> DataArray:
+    """
+    Apply a function to raster data.
+
+    Parameters
+    ----------
+    data
+        The data to process. It can be a `DataArray` or `DataTree`. If it's a `DataTree`,
+        the function is applied to the first scale (full-resolution data).
+    func
+        The function to apply to the data.
+    fn_kwargs
+        Additional keyword arguments to pass to the function `func`.
+    blockwise
+        If `True`, distributed processing will be achieved with `dask.array.map_overlap`/`dask.array.map_blocks`,
+        otherwise the function is applied to the full data. If `False`, `depth` and `chunks` are ignored.
+    depth
+        If not `None`, distributed processing will be achieved with `dask.array.map_overlap`, otherwise with
+        `dask.array.map_blocks`. Specifies the overlap between chunks, i.e. the number of elements that each chunk
+        should share with its neighbor chunks. Please see `dask.array.map_overlap` for more information on the accepted
+        values.
+    chunks
+        Passed to `dask.array.map_overlap`/`dask.array.map_blocks` as `chunks`. Ignored if `blockwise` is `False`.
+        Chunk shape of resulting blocks if the function does not preserve the data shape. If not provided, the resulting
+        array is assumed to have the same chunk structure as the first input array.
+        E.g. ( (3,), (100,100), (100,100) ).
+    c_coords
+        The channel coordinates for the output data. If not provided, the channel coordinates of the input data are
+        used. It should be specified if the function changes the number of channels.
+    dims
+        The dimensions of the output data. If not provided, the dimensions of the input data are used. It must be
+        specified if the function changes the data dimensions.
+        E.g. ('c', 'y', 'x').
+    transformations
+        The transformations of the output data. If not provided, the transformations of the input data are copied to the
+        output data. It should be specified if the function changes the data transformations.
+    kwargs
+        Additional keyword arguments to pass to `dask.array.map_overlap` or `dask.array.map_blocks`.
```

Comment:
I changed it to:

```
    kwargs
        Additional keyword arguments to pass to :func:`dask.array.map_overlap` or :func:`dask.array.map_blocks`.
        Ignored if `blockwise` is set to `False`.
```

I checked, and the link to the dask documenttion is made.

In the description of the docstring I also added:
```
    Apply a Callable to raster data.

    Applies a Callable (`func`) to raster data. If `blockwise` is set to True,
    distributed processing will be achieved with `dask.array.map_overlap`/`dask.array.map_blocks`,
    otherwise `func` is appplied to the full data.
```

Maybe this is a bit too much. We already explain it in description of `blockwise` parameter.